docker/tests

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+cd /app || (echo "Cannot find /app directory"; exit 2)
+set -xeuo pipefail
+
+if [[ "$#" -ge 1 ]] && [[ "$1" == "watch" ]]; then
+  shift
+  ptw --spool=1000 -n -c -w -- -Werror -vv "$@"
+else
+  python -m pytest -Werror -vv "$@"
+fi
```

Comment:
This was another thing that was problematic for me in the past, running tests!  How was this working for you?  Even if we remove the Docker setup in this, I'd still be happy to walk through briefly to see.